x/concentrated-liquidity/tick.go

```diff
@@ -145,6 +146,74 @@ func GetMinAndMaxTicksFromExponentAtPriceOne(exponentAtPriceOne sdk.Int) (minTic
 	return math.GetMinAndMaxTicksFromExponentAtPriceOneInternal(exponentAtPriceOne)
 }
 
+// GetTickLiquidityForRangeInBatches returns an array of liquidity depth within the given range of lower tick and upper tick.
+// The ticks within the array are always guaranteed spacing greater than then given batch size.
+func (k Keeper) GetTickLiquidityForRangeInBatches(ctx sdk.Context, poolId uint64, lowerTick, upperTick int64, batchSize uint64) ([]types.LiquidityDepth, error) {
+	liquidityDepths := []types.LiquidityDepth{}
+	store := ctx.KVStore(k.storeKey)
+	if !k.poolExists(ctx, poolId) {
+		return []types.LiquidityDepth{}, types.PoolNotFoundError{PoolId: poolId}
+	}
+
+	if upperTick < lowerTick {
+		return []types.LiquidityDepth{}, types.InvalidLowerUpperTickError{LowerTick: lowerTick, UpperTick: upperTick}
+	}
+
+	// use false for zeroForOne since we're going from lower tick -> upper tick
+	zeroForOne := false
+	swapStrategy := swapstrategy.New(zeroForOne, sdk.ZeroDec(), k.storeKey)
+
+	// we do given lower tick - 1 here to make lower tick inclusive
+	currentTick := lowerTick - 1
+
+	// iterate until we hit the upper tick given
+	// we add one here to make upper tick inclusive
+	for currentTick < upperTick+1 {
```

Comment:
question, did you have some test case to be certain that this is required in both the upward and downward direction? I ask because I think going one was is already inclusive whereas the other would require this change.